library/core/src/main/java/com/google/android/exoplayer2/video/MediaCodecVideoRenderer.java

```diff
@@ -1750,7 +1753,8 @@ public void onFrameRendered(MediaCodec codec, long presentationTimeUs, long nano
         // Stale event.
         return;
       }
-      onProcessedTunneledBuffer(presentationTimeUs);
+
+      handler.post(() -> onProcessedTunneledBuffer(presentationTimeUs));
```

Comment:
A comment should be added that the codec is not reentrant from a callback.

Reply:
This is also fixed in the platform code now. So this workaround can be behind an "if(Util.SDK_INT < 30)" to avoid the post on upcoming Android versions.

Reply:
For reference, it was fixed by https://android-review.googlesource.com/1156807.